backend/src/main/java/heartbeat/service/report/GenerateReporterService.java

```diff
@@ -50,7 +50,7 @@ public class GenerateReporterService {
 
 	private List<Map.Entry<String, List<BuildKiteBuildInfo>>> buildInfosList = new ArrayList<>();
 
-	public ReportResponse generateReporter(GenerateReportRequest request) {
+	public synchronized ReportResponse generateReporter(GenerateReportRequest request) {
```

Comment:
This lock is designed to solve the problem of the front-end calling twice at the same time